### PR TITLE
Add several spelling corrections

### PR DESCRIPTION
**Pull request by pabs3**

### CODE REVIEW

File: codespell_lib/data/dictionary.txt [hunk 1]
```diff
@@ -4434,11 +4434,70 @@ cnter->center
 co-incided->coincided
 co-ordinate->coordinate
 co-ordinates->coordinates
+coalace->coalesce
+coalacece->coalescence
```
Comment (peternewman) [inline codespell_lib/data/dictionary.txt]: I'd also suggest coalesce here.

File: codespell_lib/data/dictionary.txt [hunk 2]
```diff
@@ -6823,12 +6882,34 @@ declaratinos->declarations
 declaratiom->declaration
 declaraton->declaration
 declaratons->declarations
+declard->declared
+declarded->declared
 declaritive->declarative
 declaritively->declaratively
+declarnig->declaring
+declars->declared, declares,
```
Comment (peternewman) [inline codespell_lib/data/dictionary.txt]: Suggest these the other way round please.

File: codespell_lib/data/dictionary.txt [hunk 3]
```diff
@@ -6823,12 +6882,34 @@ declaratinos->declarations
...
+declartated->declarated
```
Comment (peternewman) [inline codespell_lib/data/dictionary.txt]: I wonder if this should also suggest declared?

File: codespell_lib/data/dictionary.txt [hunk 4]
```diff
@@ -6823,12 +6882,34 @@ declaratinos->declarations
...
+declartation->declaration
+declartations->declarations
+declartative->declarative
+declartator->declarator
+declartators->declarators
+declarted->declarated
```
Comment (peternewman) [inline codespell_lib/data/dictionary.txt]: I think this should definitely also suggest declared.

File: codespell_lib/data/dictionary.txt [hunk 5]
```diff
@@ -10575,7 +10656,10 @@ fonction->function
 fonctional->functional
 fonctionalities->functionalities
 fonctioning->functioning
+fonctionnalies->functionalities
 fonctionnalities->functionalities
+fonctionnality->functionality
+fonctionnaly->functionality, functionally,
```
Comment (peternewman) [inline codespell_lib/data/dictionary.txt]: Can you swap the suggestions, I think the latter is the more likely intention from the typo.

File: codespell_lib/data/dictionary.txt [hunk 6]
```diff
@@ -13808,6 +13892,8 @@ lavels->levels
 layed->laid
 lazer->laser
 lazyness->laziness
+lcoal->local
```
Comment (peternewman) [inline codespell_lib/data/dictionary.txt]: Also suggest coal.

File: codespell_lib/data/dictionary.txt (hunk 3 above)
Comment (peternewman) [reply codespell_lib/data/dictionary.txt]: Also Travis says this word is also a typo:
error declartated: correction declarated is an error itself

**Comment by pabs3** [comment]
Applied all requested changes.

**Pull request merged**